tools/quickstart/quickstart/utils.py

```diff
@@ -0,0 +1,143 @@
+import json
+import os
+import re
+import sys
+from pathlib import Path
+from typing import Tuple
+
+import click
+from eth_account import Account
+
+from quickstart.constants import (
+    ADDRESS_FILE_PATH,
+    BRIDGE_CONFIG_FILE_EXTERNAL,
+    KEY_DIR,
+    NETSTATS_ENV_FILE_PATH,
+    PASSWORD_FILE_PATH,
+)
+
+
+def ensure_clean_setup():
+    if os.path.isdir(KEY_DIR):
+        raise click.ClickException(
+            "The directory holding the keys already exists.\n"
+            "This should not occur during normal operation."
+        )
+
+    if os.path.isfile(PASSWORD_FILE_PATH):
+        raise click.ClickException(
+            "The password file already exists.\n"
+            "This should not occur during normal operation."
+        )
+
+
+def is_validator_account_prepared() -> bool:
+    return os.path.isfile(ADDRESS_FILE_PATH)
+
+
+def is_netstats_prepared() -> bool:
+    return (
+        os.path.isfile(NETSTATS_ENV_FILE_PATH)
+        # Ignore touched only files (docker-compose workaround)
+        and os.stat(NETSTATS_ENV_FILE_PATH).st_size != 0
+    )
+
+
+def is_bridge_prepared() -> bool:
+    return (
+        os.path.isfile(BRIDGE_CONFIG_FILE_EXTERNAL)
+        # Ignore touched only files (docker-compose workaround)
+        and os.stat(BRIDGE_CONFIG_FILE_EXTERNAL).st_size != 0
+    )
+
+
+class TrustlinesFiles:
+    def __init__(self, password_file, address_file, keystore_file):
+        self.password_file = password_file
+        self.address_file = address_file
+        self.keystore_file = keystore_file
+
+    def store(self, account, password):
+        # Instead of just copying the file, we encrypt it again since the
+        # file itself may use scrypt as key derivative function which
+        # parity can't handle
+        json_account = account.encrypt(password, kdf="pbkdf2")
+
+        os.makedirs(os.path.dirname(os.path.abspath(self.keystore_file)), exist_ok=True)
+        with open(self.keystore_file, "w") as f:
```

Comment:
Hmm. I guess that was the intention for the `ensure_clean_setup()` function. Don't you think this is enough. How would you react at this place if the file already exist? You can't just continue? Should the ensure function be removed then? Please make a concrete suggestion. Thanks! :upside_down_face:

Reply:
No, I don't think it is enough. It's easy to remove this function call or to introduce a different code path without noticing that this can have catastrophic effects. Just check if the file exists and if it does raise some exception. Even a `ValueError` should be fine if you don't want to think about a nice error message here.

Reply:
Okay, we are on the same page.
I prevent to overwrite any validator account related files. This is a replacement for the previous function, which becomes an obsolete overhead by this. Even the response to the user is more useful now in my opinion.